atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -0,0 +1,363 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import unicode_literals
+
+from atomic_reactor.plugin import PreBuildPlugin
+
+from collections import namedtuple
+import logging
+import yaml
+
+
+# Key used to store the config object in the plugin workspace
+WORKSPACE_CONF_KEY = 'reactor_config'
+
+
+def get_config(workflow):
+    """
+    Obtain configuration object
+    Does not fail
+
+    :return: ReactorConfig instance
+    """
+    try:
+        workspace = workflow.plugin_workspace[ReactorConfigPlugin.key]
+        return workspace[WORKSPACE_CONF_KEY]
+    except KeyError:
+        # The plugin did not run or was not successful: use defaults
+        return ReactorConfig()
+
+
+class ConfigBuilder(object):
```

Comment:
>YAML is easier to write and allows comments.

That's a good point, lets stick to yaml then.

Although comments are possible in json (as any other javascript code) python's interpreter is strict and doesn't allow those